blog-server-services/src/impls/rbatis_comment_service.rs

```diff
@@ -0,0 +1,75 @@
+use crate::traits::comment_service::{BaseComment, Comment, CommentService};
+use rbatis::rbatis::RBatis;
+use screw_components::dyn_result::{DError, DResult};
+
+pub fn create_rbatis_comment_service(rb: RBatis) -> Box<dyn CommentService> {
+    Box::new(RbatisCommentService { rb })
+}
+
+impl_insert!(BaseComment {}, "post_comment");
+
+impl Comment {
+    #[py_sql(
+        "
+        SELECT COUNT(1) \
+        FROM post_comment \
+        WHERE post_comment.post_id = #{post_id}
+    "
+    )]
+    async fn count_by_post_id(rb: &RBatis, post_id: &i64) -> rbatis::Result<i64> {
+        impled!()
+    }
+    #[py_sql(
+        "
+        SELECT \
+            post_comment.*, \
+            author.slug AS author_slug, \
+            author.first_name AS author_first_name, \
+            author.last_name AS author_last_name \
+        FROM post_comment \
+        JOIN author ON post_comment.author_id = author.id \
+        WHERE post_comment.post_id = #{post_id} \
+        GROUP BY post_comment.id, author.slug, author.first_name, author.last_name \
```

Comment:
Could you please check if we actually need this gtoup by stuff

Reply:
```
{
  "failure": {
    "identifier": "COMMENTS_DATABASE_ERROR",
    "reason": "database error: db:Notice { storage: b\"SERROR\\0VERROR\\0C42803\\0Mcolumn \\\"author.slug\\\" must appear in the GROUP BY clause or be used in an aggregate function\\0P24\\0Fparse_agg.c\\0L1428\\0Rcheck_ungrouped_columns_walker\\0\\0\", severity: Error, message: (22, 113), code: (15, 20) }"
  }
}
```

Reply:
thx MY BOI